src/debug/daccess/request_svr.cpp

```diff
@@ -227,7 +227,9 @@ ClrDataAccess::EnumSvrGlobalMemoryRegions(CLRDataEnumMemoryFlags flags)
         size_t gen_table_size = g_gcDacGlobals->generation_size * (*g_gcDacGlobals->max_gen + 2);
         DacEnumMemoryRegion(dac_cast<TADDR>(pHeap), sizeof(dac_gc_heap));
         DacEnumMemoryRegion(dac_cast<TADDR>(pHeap->finalize_queue), sizeof(dac_finalize_queue));
-        DacEnumMemoryRegion(dac_cast<TADDR>(pHeap->generation_table), gen_table_size);
+
+        TADDR taddrTable = dac_cast<TADDR>(pHeap) + offsetof(dac_gc_heap, generation_table);
```

Comment:
What is different about the new line vs. the old one? If they compute different values does that mean the same issue exists in the DacEnumMemoryRegion(pHeap->finalize_queue) call that preceeds this?

Reply:
The original code was throwing an invalid memory exception because generation_table is an array and the dac_cast macro didn't seem to handle it correctly.